tsconfig.typecheck.json

```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "./tsconfig.json",
+  "exclude": ["**/dist/**", "**/__coverage__/**", "packages/starlight/components.ts"]
```

Comment:
Could it make sense to exclude these in the default tsconfig instead of adding the custom one?

Reply:
The issue is that we only want to exclude `packages/starlight/components.ts` during type-checking and not while working on Starlight with for example the Astro extension in Visual Studio Code.
`**/dist/**"` and `**/__coverage__/**` are already part of the default tsconfig but unfortunately, when using `extends` in a `tsconfig.json` file, arrays are not merged.